airbyte-integrations/connectors/source-salesforce/unit_tests/integration/test_bulk_stream.py

```diff
@@ -65,29 +71,225 @@ def tearDown(self) -> None:
+            JobInfoResponseBuilder().with_id(_JOB_ID).with_state("JobComplete").build(),
+        )
+        self._http_mocker.get(
+            HttpRequest(f"{_BASE_URL}/jobs/query/{_JOB_ID}/results"),
+            HttpResponse(f"{_A_FIELD_NAME}\nfield_value"),
+        )
+        self._http_mocker._mock_request_method(  # FIXME to add DELETE method in airbyte_cdk tests
+            "delete",
+            HttpRequest(f"{_BASE_URL}/jobs/query/{_JOB_ID}"),
+            [
+                _RETRYABLE_RESPONSE,
+                HttpResponse(""),
+            ],
+        )
+
+        output = read(_STREAM_NAME, SyncMode.full_refresh, self._config)
+
+        assert output.get_stream_statuses(_STREAM_NAME)[-1] == AirbyteStreamStatus.COMPLETE
+
+    def test_given_non_retryable_error_on_delete_job_result_when_read_then_fail_to_sync(self):
+        """
+        This is interesting: right now, we retry with the same policies has the other requests but it seems fair to just be a best effort,
+        catch everything and not retry
+        """
+        given_stream(self._http_mocker, _BASE_URL, _STREAM_NAME, SalesforceDescribeResponseBuilder().field(_A_FIELD_NAME))
+        self._http_mocker.post(
+            self._make_full_job_request([_A_FIELD_NAME]),
+            JobCreateResponseBuilder().with_id(_JOB_ID).build(),
+        )
+        self._http_mocker.get(
+            HttpRequest(f"{_BASE_URL}/jobs/query/{_JOB_ID}"),
+            JobInfoResponseBuilder().with_id(_JOB_ID).with_state("JobComplete").build(),
+        )
+        self._http_mocker.get(
+            HttpRequest(f"{_BASE_URL}/jobs/query/{_JOB_ID}/results"),
+            HttpResponse(f"{_A_FIELD_NAME}\nfield_value"),
+        )
+        self._http_mocker._mock_request_method(  # FIXME to add DELETE method in airbyte_cdk tests
+            "delete",
+            HttpRequest(f"{_BASE_URL}/jobs/query/{_JOB_ID}"),
+            HttpResponse("", 429),
+        )
+
+        output = read(_STREAM_NAME, SyncMode.full_refresh, self._config)
+
+        assert output.get_stream_statuses(_STREAM_NAME)[-1] == AirbyteStreamStatus.INCOMPLETE
+
     def test_given_incremental_when_read_then_create_job_and_extract_records_from_result(self) -> None:
         start_date = (_NOW - timedelta(days=10)).replace(microsecond=0)
         first_upper_boundary = start_date + timedelta(days=7)
         self._config.start_date(start_date).stream_slice_step("P7D")
-        given_stream(self._http_mocker, _BASE_URL, _STREAM_NAME, SalesforceDescribeResponseBuilder().field(_A_FIELD_NAME).field(_CURSOR_FIELD, "datetime"))
-        self._create_sliced_job(start_date, first_upper_boundary, "first_slice_job_id", self._generate_csv([_A_FIELD_NAME, _CURSOR_FIELD], count=2))
-        self._create_sliced_job(first_upper_boundary, _NOW, "second_slice_job_id", self._generate_csv([_A_FIELD_NAME, _CURSOR_FIELD], count=1))
+        given_stream(self._http_mocker, _BASE_URL, _STREAM_NAME, _INCREMENTAL_SCHEMA_BUILDER)
```

Comment:
This was nice, I like constants.
nit: maybe something similar for URL paths?

Reply:
For stripe, we have a [request builder](https://github.com/airbytehq/airbyte/blob/f23c2e6d9254ebeceba975c8b32406297d1b2091/airbyte-integrations/connectors/source-stripe/unit_tests/integration/request_builder.py) we tried experimenting with. I think it has value in terms of maintenance i.e. that if the endpoint change but the logic for the test is the same, we just have to update in one place. However, I don't think it adds much in terms of readability. 

Since this comment is nit and this is already spread, I would leave it as is and if the endpoint change, we can add the builder pattern

Reply:
Awesome, I agree, feel free to always resolve my nit comments.